online-shop/src/app/list-products-detail/list-products-detail.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, OnInit } from '@angular/core';
+import {ProductService} from "../services/product.service";
+import {Product} from "../product";
+import {FormBuilder, FormGroup, Validators} from "@angular/forms";
+import {ProductModel} from "../productModel";
+import {AuthenticationService} from "../services/authentication.service";
+
+@Component({
+  selector: 'app-list-products-detail',
+  templateUrl: './list-products-detail.component.html',
+  styleUrls: ['./list-products-detail.component.scss']
+})
+export class ListProductsDetailComponent implements OnInit {
+
+  products: Product[]=[]
+  formValue !: FormGroup;
+
+  productModelObj: ProductModel = new ProductModel();
+
+  constructor(private productService: ProductService, private formBuilder: FormBuilder, public authenticationService: AuthenticationService) {
+    this.createForm();
+
+  }
+
+  ngOnInit(): void {
+    this.get();
+
+    this.formValue = this.formBuilder.group({
+      productName: [''],
+      productCategory: [''],
+      productPrice: [''],
+      productImage: [''],
+      productDescription: ['']
+    })
+  }
+
+  createForm() {
+    this.formValue = this.formBuilder.nonNullable.group({
+      productName: ['', Validators.required ],
+      productPrice: ['', Validators.required ]
+    });
+  }
+
+  get(): void {
+    this.productService.getProducts().subscribe(data => {this.products = data;})
```

Comment:
**suggestion:** take care to unsubscribe to avoid memory leaks inside your code. https://www.thisdot.co/blog/best-practices-for-managing-rxjs-subscriptions